src/webgpu/api/validation/compute_pipeline.spec.ts

```diff
@@ -259,3 +259,254 @@ Tests calling createComputePipeline(Async) validation for compute workgroup_size
       size[2] <= t.device.limits.maxComputeWorkgroupSizeZ;
     t.doCreateComputePipelineTest(isAsync, _success, descriptor);
   });
+
+g.test('overrides,identifier')
+  .desc(
+    `
+Tests calling createComputePipeline(Async) validation for overridable constants identifiers.
+`
+  )
+  .params(u =>
+    u //
+      .combine('isAsync', [true, false])
+      .combineWithParams([
+        { constants: {}, _success: true },
+        { constants: { c0: 0 }, _success: true },
+        { constants: { c0: 0, c1: 1 }, _success: true },
+        { constants: { c9: 0 }, _success: false },
+        { constants: { 1: 0 }, _success: true },
+        { constants: { c3: 0 }, _success: false }, // pipeline constant id is specified for c3
+        { constants: { 2: 0 }, _success: false },
+        { constants: { 1000: 0 }, _success: true },
+        { constants: { 9999: 0 }, _success: false },
+        { constants: { 1000: 0, c2: 0 }, _success: false },
+      ] as { constants: Record<string, GPUPipelineConstantValue>; _success: boolean }[])
```

Comment:
nit: would simply `as const` work instead? (I think it should)

Reply:
Turns out no. This is the error log from vscode intellisense. `as const` seems to infer different types for each constants (dictionary with different keys)

```
Argument of type '{ layout: "auto"; compute: { module: GPUShaderModule; entryPoint: string; constants: { readonly x?: undefined; readonly y?: undefined; readonly z?: undefined; } | { readonly x: 0; readonly y: 0; readonly z: 0; } | { ...; } | { ...; } | { ...; }; }; }' is not assignable to parameter of type 'GPUComputePipelineDescriptor'.
```